### ISSUE TITLE
undefined reference to `clCreateCommandQueueWithProperties'

### ISSUE THREAD

**Issue by jszczerb** [fireice-uk/xmr-stak-amd#40]
os:  Ubuntu 17.04

Built and currently running the v1.0.0-1.3.1 release just fine... but now I'm trying to build the  v1.1.0-1.4.0 release

and get the following:  

[ 47%] Linking CXX executable bin/xmr-stak-amd
libxmr-stak-amd-c.a(gpu.c.o): In function `InitOpenCLGpu':
gpu.c:(.text+0x56a): undefined reference to `clCreateCommandQueueWithProperties'
collect2: error: ld returned 1 exit status

ideas?

**Comment by fireice-uk** [comment]
@psychocrypt any reason why we would get opencl linking errors with the cmake changes?

**Comment by psychocrypt** [comment]
@jszczerb Please post your CMake output and run `cmake -L` in the build directory and if you used git than `git log` in the source code directory. Please post all outputs.

**Comment by jszczerb** [comment]
[gitlog.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1049914/gitlog.txt)

**Comment by jszczerb** [comment]
[cmakelog.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1049916/cmakelog.txt)

**Comment by psychocrypt** [comment]
Have you build the new miner in a clean directory without any temporary cmake files from older compiles?
During the link process the opencl library is missing.
Please compile with

```
make VERBOSE=1
```
and post the output.

**Comment by jszczerb** [comment]
Yes it was in a clean directory.  I will do it again with verbose=1

Jeff Szczerbinski, CCIE#4052 Emeritus
jeff@201grandavenue.com
whatsapp:  4143508658
skype:  jszczerb

**Comment by eronquillo** [comment]
Hi, would there be any environment variable that need to be assigned to complete the build?  Also, would having xmr-stak-nvidia (video and cuda drivers) interfere with the amd build?  I'm running Ubuntu 16.04 with the nvidia xmr-stak with no problems (had to build with LLVM instead of GCC)

**Comment by GanjaPowered** [comment]
Hi, I have exactly the same problem on Ubuntu 16.04
cmake find OpenCL:

> -- Found OpenCL: /usr/lib/x86_64-linux-gnu/libOpenCL.so (found version "2.0")

But during make I get undefined reference:

> [100%] Linking CXX executable bin/xmr-stak-amd
libxmr-stak-amd-c.a(gpu.c.o) : Dans la fonction « InitOpenCLGpu » :
gpu.c:(.text+0x1ad) : référence indéfinie vers « clCreateCommandQueueWithProperties »

Full install results: [terminal.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1244983/terminal.txt)
Make verbose: [make verbose.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1244987/make.verbose.txt)
cmake -L: [cmake l.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1244991/cmake.l.txt)

Thanks for helping

**Comment by psychocrypt** [comment]
@GanjaPowered it could be that you have installed two olencl libraries and the miner selected the wrong library

**Comment by GanjaPowered** [comment]
@psychocrypt Thanks a lot, linked /usr/lib/libOpenCL.so to /opt/AMDAPPSDK-3.0/lib/x86_64/sdk/libOpenCL.so.1 and now it works

**Comment by psychocrypt** [comment]
Could you shortly at a description how you solved the wrong linking issue? This will help the next user with the same issue.

**Comment by GanjaPowered** [comment]
@psychocrypt I previously installed nvidia GPUs on my rig so I had two versions of libOpenCL.so on my system. 
**/usr/lib/libOpenCL.so** was a symbolic link to **/usr/lib/x86_64-linux-gnu/libOpenCL.so**, which was the **nvidia one** (I guess, the file size was different from **/opt/AMDAPPSDK-3.0/lib/x86_64/sdk/libOpenCL.so.1**).

So I did a backup of **/usr/lib/x86_64-linux-gnu/libOpenCL.so** before removing it and then replace it with a symbolic link to the Radeon library.

> sudo cp /usr/lib/x86_64-linux-gnu/libOpenCL.so /usr/lib/x86_64-linux-gnu/libOpenCL.so.old
> sudo rm /usr/lib/x86_64-linux-gnu/libOpenCL.so
> sudo ln -s /usr/lib/x86_64-linux-gnu/libOpenCL.so /opt/AMDAPPSDK-3.0/lib/x86_64/sdk/libOpenCL.so.1

It's probably not the most clean way, but it's efficient.
